### ISSUE TITLE
Diff results show a diff where there is none

### ISSUE THREAD

**Issue by DotFreelance** [andreyvit/json-diff#12]
![diff-problem](https://cloud.githubusercontent.com/assets/6014366/6006230/bc548536-aac5-11e4-9606-35c13603cf0d.png)
As you can see by the screenshot, the diff tool seems to be up and straight missing. I don't know what the tool is missing here, but these two files are the same ( they're actually the exact same content, used for testing purposes. )

I'm not really able to research and build a PR, sadly. I'm going to have to look for another tool. If anyone can figure this one out that would be awesome. :)

Edit: Just thought I'd mention that this is a series of objects in an array that's nested in an object which is a part of a series nested in an array, etc.

**Comment by andreyvit** [comment]
Hey, can you please attach a sample file?

**Comment by stampycode** [comment]
The following diff fails:

    var json = {
      "list":  [
        {
          "a": ""
        },
        {
          "a": ""
        }
      ]
    };
    diff(json, json);

produces:

       {
         list: [
           ...
      -    {
      -      a: ""
      -    }
      +    {
      +      a: ""
      +    }
         ]
       }

**Comment by hellboy81** [comment]
Have the same issue

**Comment by Asuza** [comment]
@andreyvit, it's been a year now... any update on this?

**Comment by hellboy81** [comment]
fix it yourself

**Comment by marcellsimon** [comment]
This is still an issue, and it got even more weird, but I created a small recursive function to handle this:

`import jsonDiff from 'json-diff'

export default function differ(obj1, obj2) {
  let diff = jsonDiff.diff(obj1, obj2)
  diff = clear(diff)
  return diff
}


function clear(diff) {
  //console.log('=---------------------------=')
  //console.log('=---------------------------=')
  //console.log(JSON.stringify(diff))
  //console.log('==')
  if (Array.isArray(diff)) {
    //console.log('array')
    let newArray = []
    for (const item of diff) {
      if (item.length !== 1 || item[0] !== " ") {
        newArray.push(clear(item))
      }
    }
    
    newArray = newArray.filter(item => item)
    
    if (newArray.length === 0) {
      //console.log('but empty')
      return undefined
    } else {
      
      //console.log('with items')
      return newArray
    }
  } else if (typeof diff === 'object') {
    //console.log('object')
    if (Object.keys(diff).length) {
      const newObject = {}
      for (const key of Object.keys(diff)) {
        const objDiff = clear(diff[key])
        if (objDiff) {
          newObject[key] = objDiff
        }
      }
      
      if (Object.keys(newObject).length) {
        //console.log('with items')
        return newObject
      } else {
        //console.log('but empty')
        return undefined
      }
    } else {
      return undefined
    }
  } else {
    //console.log('scalar')
    if (diff && `${diff}`.trim() && diff !== "~") {
      //console.log('ok')
      return diff
    } else {
      //console.log('nok')
      return undefined
    }
  }
}`

**Comment by ewoudenberg** [comment]
This has been fixed in 0.6.0:
```
$ ./node_modules/.bin/json-diff --full <(echo '[{"a":""},{"a":""}]') <(echo '[{"a":""},{"a":""}]')
 [
   {
     a: ""
   }
   {
     a: ""
   }
 ]
$ ./node_modules/.bin/json-diff  <(echo '[{"a":""},{"a":""}]') <(echo '[{"a":""},{"a":""}]')
$
```
